src/metpy/plots/declarative.py

```diff
@@ -479,13 +514,29 @@ def draw(self):
                 self.layers_edgecolor *= len(self.layers)
             if len(self.layers) > len(self.layers_linewidth):
                 self.layers_linewidth *= len(self.layers)
+            if len(self.layers) > len(self.layers_linestyle):
+                self.layers_linestyle *= len(self.layers)
+            if len(self.layers) > len(self.layers_zorder):
+                self.layers_zorder *= len(self.layers)
+            if len(self.layers) > len(self.layers_alpha):
+                self.layers_alpha *= len(self.layers)
             for i, feat in enumerate(self._layer_features):
-                if self.layers[i] in ['', 'land', 'lake', 'river']:
+                color = self.layers_edgecolor[i]
+                if self.layers[i] in ['', 'land', 'ocean']:
                     color = 'face'
-                else:
-                    color = self.layers_edgecolor[i]
+                if self.layers_edgecolor[i] in ['water', 'land', 'land_alt1']:
+                    color = get_cartopy_color(self.layers_edgecolor[i])
                 width = self.layers_linewidth[i]
-                self.ax.add_feature(feat, edgecolor=color, linewidth=width)
+                style = self.layers_linestyle[i]
+                zorder = self.layers_zorder[i]
+                alpha = self.layers_alpha[i]
+                if zorder is not None:
+                    self.ax.add_feature(feat, edgecolor=color,
+                                        linewidth=width, linestyle=style, zorder=zorder,
+                                        alpha=alpha)
+                else:
+                    self.ax.add_feature(feat, edgecolor=color,
+                                        linewidth=width, linestyle=style, alpha=alpha)
```

Comment:
Less repetition this way:
```suggestion
                kwargs = {'zorder': zorder} if zorder is not None else {}
                self.ax.add_feature(feat, edgecolor=color, linewidth=width,
                                    linestyle=style, alpha=alpha, **kwargs)
```